core/domain/collection_services_test.py

```diff
@@ -116,6 +116,49 @@ def test_get_collection_titles_and_categories(self):
                     }
                 })
 
+    def test_get_acquired_skills_of_user_given_collection_id(self):
+        """Tests get_acquired_skills_of_user_given_collection_id method."""
+        collection_id = 'col1'
+        exp_id = '0_exploration_id'
+        owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+        # Create a new collection and exploration.
+        self.save_new_valid_collection(
+            collection_id, owner_id, exploration_id=exp_id)
+
+        # Add a skill.
+        collection_services.update_collection(
+            owner_id, collection_id, [{
+                'cmd': collection_domain.CMD_ADD_COLLECTION_SKILL,
+                'name': 'test'
+            }], 'Add a new skill')
+        collection = collection_services.get_collection_by_id(
+            collection_id)
+        skill_id = collection.get_skill_id_from_skill_name('test')
+        collection_node = collection.get_node(exp_id)
+        collection_node.update_acquired_skill_ids([skill_id])
+
+        # Update skill.
```

Comment:
I don't think I understand why you need this step. Is it really necessary for the test?

Reply:
yes, update_collection makes changes to domain object but will not save it to collection model. In order to make changes(add skill to collection), I am doing this.

Reply:
Oh, I see what you're doing now (on a closer reading). Could you maybe change this comment to something more explicit like "Update the acquired skill IDs for the exploration."? That would convey the intent better.

Reply:
Updated.